From 85a6e09b3e437a0bc17a7ec01273e647e5a2af50 Mon Sep 17 00:00:00 2001
From: ryang <ryang@hach.com>
Date: Tue, 22 May 2018 09:33:04 -0600
Subject: [PATCH] Use 14x16 font

---
 src/drivers/genfont.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/drivers/genfont.c b/src/drivers/genfont.c
index 6ca253a..34d0740 100644
--- a/src/drivers/genfont.c
+++ b/src/drivers/genfont.c
@@ -19,7 +19,7 @@
 /* compiled in fonts*/
 extern MWCFONT font_winFreeSansSerif11x13;	/* new MWFONT_SYSTEM_VAR (was MWFONT_GUI_VAR)*/
 extern MWCFONT font_X6x13;			/* MWFONT_SYSTEM_FIXED (should be ansi)*/
-/*extern MWCFONT font_winFreeSystem14x16;*/	/* deprecated MWFONT_SYSTEM_VAR*/
+extern MWCFONT font_winFreeSystem14x16;	/* deprecated MWFONT_SYSTEM_VAR*/
 /*extern MWCFONT font_rom8x16;*/		/* deprecated MWFONT_OEM_FIXED*/
 /*extern MWCFONT font_rom8x8, font_X5x7;*/	/* unused*/
 
@@ -56,10 +56,10 @@ static MWFONTPROCS fontprocs = {
 
 /* first font is default font*/
 MWCOREFONT gen_fonts[NUMBER_FONTS] = {
-	{&fontprocs, 0, 0, 0, MWFONT_SYSTEM_VAR,   &font_winFreeSansSerif11x13},
+	{&fontprocs, 0, 0, 0, MWFONT_SYSTEM_VAR,   &font_winFreeSystem14x16},
 	{&fontprocs, 0, 0, 0, MWFONT_SYSTEM_FIXED, &font_X6x13},
 	/* deprecated redirections for the time being*/
-	{&fontprocs, 0, 0, 0, "Helvetica",         &font_winFreeSansSerif11x13}, /* redirect*/
+	{&fontprocs, 0, 0, 0, "Helvetica",         &font_winFreeSystem14x16}, /* redirect*/
 	{&fontprocs, 0, 0, 0, "Terminal",          &font_X6x13}	/* redirect*/
 };
 
-- 
2.11.0

