From 39f836d591213832de51ad71378e27c06a0985f4 Mon Sep 17 00:00:00 2001
From: Devang Patel <dpatel@hach.com>
Date: Wed, 10 Jan 2018 13:18:35 -0700
Subject: [PATCH 1/9] Adding hach-r1701 as carrier board in to OTP kernel

---
 drivers/char/fsl_otp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/char/fsl_otp.c b/drivers/char/fsl_otp.c
index ba04291397ad..3502598d3302 100644
--- a/drivers/char/fsl_otp.c
+++ b/drivers/char/fsl_otp.c
@@ -799,6 +799,7 @@ static int fsl_register_carrierboard(void) {
 		"digi,ccimx6sbc",
 		"digi,ccimx6ulstarter",
 		"digi,ccimx6ulsbc",
+        "hach,r1701"
 	};
 
 	for (i = 0; i < ARRAY_SIZE(compat_machines); i++) {
-- 
2.11.0

