From 66d12a278b8a802ca711448f4b3047ac7bb6d18f Mon Sep 17 00:00:00 2001
From: Devang Patel <dpatel@hach.com>
Date: Thu, 16 May 2019 10:35:28 -0600
Subject: [PATCH 03/10] Fix-for-matrix-keypad-column-gpio-scanning-issue

---
 drivers/input/keyboard/matrix_keypad.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/input/keyboard/matrix_keypad.c b/drivers/input/keyboard/matrix_keypad.c
index c04559a232f7..cb5187ad5018 100644
--- a/drivers/input/keyboard/matrix_keypad.c
+++ b/drivers/input/keyboard/matrix_keypad.c
@@ -53,11 +53,15 @@ static void __activate_col(const struct matrix_keypad_platform_data *pdata,
 	bool level_on = !pdata->active_low;
 
 	if (on) {
-		gpio_direction_output(pdata->col_gpios[col], level_on);
+		//gpio_direction_output(pdata->col_gpios[col], level_on);
+		gpio_set_value(pdata->col_gpios[col], level_on);
 	} else {
+		/*
 		gpio_set_value_cansleep(pdata->col_gpios[col], !level_on);
 		if (!pdata->drive_inactive_cols)
 			gpio_direction_input(pdata->col_gpios[col]);
+		*/
+		gpio_set_value(pdata->col_gpios[col], !level_on);
 	}
 }
 
@@ -82,8 +86,12 @@ static void activate_all_cols(const struct matrix_keypad_platform_data *pdata,
 static bool row_asserted(const struct matrix_keypad_platform_data *pdata,
 			 int row)
 {
+	/*
 	return gpio_get_value_cansleep(pdata->row_gpios[row]) ?
 			!pdata->active_low : pdata->active_low;
+	*/
+	return gpio_get_value(pdata->row_gpios[row]) ?
+			!pdata->active_low : pdata->active_low;
 }
 
 static void enable_row_irqs(struct matrix_keypad *keypad)
-- 
2.17.1

