From f07236405265b5ef2bf7367da396ae0fa161bf17 Mon Sep 17 00:00:00 2001
From: Devang Patel <dpatel@hach.com>
Date: Thu, 10 Sep 2020 11:21:37 -0600
Subject: [PATCH 1/1] Color LCD driver patch

---
 drivers/video/fbdev/mxsfb.c | 71 +++++++++++++++++++++++++++++++++++++
 1 file changed, 71 insertions(+)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 33750af90c05..b0ce6a7b4d25 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -63,6 +63,10 @@
 #include <video/videomode.h>
 #include <linux/uaccess.h>
 
+// HACH - For LCD Control lines initialization.
+#include <linux/gpio.h>
+#include <linux/delay.h>
+
 #include "mxc/mxc_dispdrv.h"
 
 #define REG_SET	4
@@ -268,6 +272,11 @@ struct mxsfb_info {
 #ifdef CONFIG_FB_MXC_OVERLAY
 	struct mxsfb_layer overlay;
 #endif
+
+    // HACH - LCD control lines GPIO pins.
+    int lcd_pwr_en;
+    int lcd_standby;
+    int lcd_reset;
 };
 
 #define mxsfb_is_v3(host) (host->devdata->ipversion == 3)
@@ -2231,6 +2240,47 @@ static int mxsfb_probe(struct platform_device *pdev)
 		return -ENOMEM;
 	}
 
+    // HACH - Initialize LCD Control lines.
+    host->lcd_pwr_en                = of_get_named_gpio(pdev->dev.of_node, "gpios", 0);    /* LCD_PWR_EN&  */
+    host->lcd_reset                 = of_get_named_gpio(pdev->dev.of_node, "gpios", 1);    /* LCD_RESET&   */
+    host->lcd_standby               = of_get_named_gpio(pdev->dev.of_node, "gpios", 2);    /* LCD_STANDBY& */
+
+    /* Enable all level shifter and turn on power to them */
+    ret = devm_gpio_request_one(&pdev->dev, host->lcd_pwr_en, GPIOF_OUT_INIT_LOW, "LCD_PWR_EN");
+    if(ret < 0){
+        dev_err(&pdev->dev, "Failed to request LCD_PWR_EN gpio pin\n");
+        return -ENODEV;
+    }
+    else{
+        gpio_export(host->lcd_pwr_en, 1);
+        mdelay(50);
+    }
+
+    /* configure and generate reset */
+    ret = devm_gpio_request_one(&pdev->dev, host->lcd_reset, GPIOF_OUT_INIT_LOW, "LCD_RESET");
+    if(ret < 0){
+        dev_err(&pdev->dev, "Failed to request LCD_RESET gpio pin\n");
+        return -ENODEV;
+    }
+    else{
+        gpio_export(host->lcd_reset, 1);
+        mdelay(20);
+        gpio_set_value(host->lcd_reset, 1);
+        mdelay(10);
+    }
+
+    /* Configure and take LCD out of standby */
+    ret = devm_gpio_request_one(&pdev->dev, host->lcd_standby, GPIOF_OUT_INIT_HIGH, "LCD_STANDBY");
+    if(ret < 0){
+        dev_err(&pdev->dev, "Failed to request LCD_STANDBY gpio pin\n");
+        return -ENODEV;
+    }
+    else{
+        gpio_export(host->lcd_standby, 1);
+        gpio_set_value(host->lcd_standby, 1);
+    }
+    // HACH - Initialization ends.
+
 	fb_info = framebuffer_alloc(0, &pdev->dev);
 	if (!fb_info) {
 		dev_err(&pdev->dev, "Failed to allocate fbdev\n");
@@ -2389,6 +2439,15 @@ static int mxsfb_remove(struct platform_device *pdev)
 	unregister_framebuffer(fb_info);
 	mxsfb_free_videomem(host);
 
+    // HACH - De-init contol lines.
+    gpio_set_value(host->lcd_standby, 0);
+    mdelay(10);
+    gpio_set_value(host->lcd_reset, 0);
+    gpio_set_value(host->lcd_pwr_en, 1);
+    devm_gpio_free(&pdev->dev, host->lcd_standby);
+    devm_gpio_free(&pdev->dev, host->lcd_reset);
+    devm_gpio_free(&pdev->dev, host->lcd_pwr_en);
+
 	platform_set_drvdata(pdev, NULL);
 
 	devm_kfree(&pdev->dev, fb_info->pseudo_palette);
@@ -2417,6 +2476,9 @@ static int mxsfb_runtime_suspend(struct device *dev)
 {
 	struct mxsfb_info *host = dev_get_drvdata(dev);
 
+    // HACH - Put LCD standby control line in to standby.
+    gpio_set_value(host->lcd_standby, 0);
+
 	if (host->devdata->flags & MXSFB_FLAG_BUSFREQ)
 		release_bus_freq(BUS_FREQ_HIGH);
 
@@ -2432,6 +2494,9 @@ static int mxsfb_runtime_resume(struct device *dev)
 {
 	struct mxsfb_info *host = dev_get_drvdata(dev);
 
+    // HACH - Put LCD standby control line out of standby.
+    gpio_set_value(host->lcd_standby, 1);
+
 	if (host->devdata->flags & MXSFB_FLAG_BUSFREQ)
 		request_bus_freq(BUS_FREQ_HIGH);
 
@@ -2452,6 +2517,9 @@ static int mxsfb_suspend(struct device *pdev)
 	struct fb_info *fb_info = host->fb_info;
 	int saved_blank;
 
+    // HACH - Put LCD standby control line in to standby.
+    gpio_set_value(host->lcd_standby, 0);
+
 	console_lock();
 	mxsfb_overlay_suspend(host);
 	fb_set_suspend(fb_info, 1);
@@ -2472,6 +2540,9 @@ static int mxsfb_resume(struct device *pdev)
 
 	pinctrl_pm_select_default_state(pdev);
 
+    // HACH - Put LCD standby control line out of standby.
+    gpio_set_value(host->lcd_standby, 1);
+
 	console_lock();
 	mxsfb_overlay_resume(host);
 	mxsfb_blank(host->restore_blank, fb_info);
-- 
2.17.1

