From 9f19a57fb2608be63d9be9933c69bb1ba7ed64a0 Mon Sep 17 00:00:00 2001
From: Devang Patel <dpatel@hach.com>
Date: Wed, 25 Oct 2017 14:24:27 -0600
Subject: [PATCH] Adding sharp mip display support.

Signed-off-by: Devang Patel <dpatel@hach.com>
---
 drivers/video/fbdev/Kconfig | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/video/fbdev/Kconfig b/drivers/video/fbdev/Kconfig
index 0e1d4b1..a75533f 100644
--- a/drivers/video/fbdev/Kconfig
+++ b/drivers/video/fbdev/Kconfig
@@ -2484,3 +2484,17 @@ config FB_SSD1307
 	help
 	  This driver implements support for the Solomon SSD1307
 	  OLED controller over I2C.
+
+config FB_LS0XXB7DXXX
+	tristate "Sharp LS027B7DH01-2.7 inch and LS032B7DD02-3.2 inch mip monochrome LCD framebuffer support"
+	depends on FB && SPI
+	depends on OF
+	depends on GPIOLIB
+	select FB_SYS_FOPS
+	select FB_SYS_FILLRECT
+	select FB_SYS_COPYAREA
+	select FB_SYS_IMAGEBLIT
+	select FB_DEFERRED_IO
+	help
+	  This driver implements support for the Sharp LS027B7DH01, LS032B7DD02
+	  LCD controller over SPI.
-- 
2.7.4

