From 04da6ea00cfcd01aaddb35ac56f389ed4f74bdba Mon Sep 17 00:00:00 2001
From: Devang Patel <dpatel@hach.com>
Date: Wed, 7 Feb 2018 11:33:10 -0700
Subject: [PATCH 2/2] Addign rev 1 board support

---
 board/digi/r1701/r1701.c | 95 +++++++++++++++++++++++++-----------------------
 include/configs/r1701.h  |  4 +-
 2 files changed, 52 insertions(+), 47 deletions(-)

diff --git a/board/digi/r1701/r1701.c b/board/digi/r1701/r1701.c
index 3d58ea2856..126acbb0e0 100644
--- a/board/digi/r1701/r1701.c
+++ b/board/digi/r1701/r1701.c
@@ -102,13 +102,15 @@ static iomux_v3_cfg_t const ext_gpios_pads[] = {
 #endif

 /* micro SD */
-static iomux_v3_cfg_t const usdhc2_pads[] = {
-	MX6_PAD_CSI_VSYNC__USDHC2_CLK | MUX_PAD_CTRL(USDHC_PAD_CTRL),
-	MX6_PAD_CSI_HSYNC__USDHC2_CMD | MUX_PAD_CTRL(USDHC_PAD_CTRL),
-	MX6_PAD_CSI_DATA00__USDHC2_DATA0 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
-	MX6_PAD_CSI_DATA01__USDHC2_DATA1 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
-	MX6_PAD_CSI_DATA02__USDHC2_DATA2 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
-	MX6_PAD_CSI_DATA03__USDHC2_DATA3 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
+static iomux_v3_cfg_t const usdhc1_pads[] = {
+	MX6_PAD_SD1_CLK__USDHC1_CLK     | MUX_PAD_CTRL(USDHC_PAD_CTRL),
+	MX6_PAD_SD1_CMD__USDHC1_CMD     | MUX_PAD_CTRL(USDHC_PAD_CTRL),
+	MX6_PAD_SD1_DATA0__USDHC1_DATA0 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
+	MX6_PAD_SD1_DATA1__USDHC1_DATA1 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
+	MX6_PAD_SD1_DATA2__USDHC1_DATA2 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
+	MX6_PAD_SD1_DATA3__USDHC1_DATA3 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
+	/* Power enable line */
+    MX6_PAD_CSI_DATA01__GPIO4_IO22 | MUX_PAD_CTRL(NO_PAD_CTRL),
 };

 #ifdef CONFIG_SYS_I2C_MXC
@@ -145,10 +147,13 @@ static iomux_v3_cfg_t const fec1_pads[] = {
 	MX6_PAD_ENET1_RX_ER__ENET1_RX_ER | MUX_PAD_CTRL(ENET_PAD_CTRL),
 	MX6_PAD_ENET1_RX_EN__ENET1_RX_EN | MUX_PAD_CTRL(ENET_PAD_CTRL),
 	/*
-	 * GPIO3_IO2 is used as PHY reset in Starter Board v1 and as PHY power
-	 * enable on Starter Board v2
+	 * GPIO3_IO23 is used as PHY reset in r1701 v1 board.
 	 */
-	MX6_PAD_LCD_HSYNC__GPIO3_IO02 | MUX_PAD_CTRL(NO_PAD_CTRL),
+	MX6_PAD_LCD_DATA18__GPIO3_IO23 | MUX_PAD_CTRL(NO_PAD_CTRL),
+	/*
+     * GPIO3_IO22 is used as power enable control pin for entire Ethernet module.
+     */
+    MX6_PAD_LCD_DATA17__GPIO3_IO22 | MUX_PAD_CTRL(NO_PAD_CTRL),
 };

 static void setup_iomux_fec(void)
@@ -164,7 +169,7 @@ static void setup_iomux_uart(void)

 #ifdef CONFIG_FSL_ESDHC
 static struct fsl_esdhc_cfg usdhc_cfg[] = {
-	{USDHC2_BASE_ADDR, 0, 4},
+	{USDHC1_BASE_ADDR, 0, 4},
 };

 int mmc_get_env_devno(void)
@@ -196,19 +201,22 @@ int board_mmc_getcd(struct mmc *mmc)

 int board_mmc_init(bd_t *bis)
 {
-	int i, ret;
+	int i, ret, sdcard_power_pin;

 	/*
 	 * According to the board_mmc_init() the following map is done:
 	 * (U-boot device node)    (Physical Port)
-	 * mmc0                    USDHC2
+	 * mmc0                    USDHC1
 	 */
 	for (i = 0; i < CONFIG_SYS_FSL_USDHC_NUM; i++) {
 		switch (i) {
 		case 0:
-			imx_iomux_v3_setup_multiple_pads(
-				usdhc2_pads, ARRAY_SIZE(usdhc2_pads));
-			usdhc_cfg[0].sdhc_clk = mxc_get_clock(MXC_ESDHC2_CLK);
+			imx_iomux_v3_setup_multiple_pads(usdhc1_pads, ARRAY_SIZE(usdhc1_pads));
+
+            /* Enable SD Card power - Active low signal */
+            sdcard_power_pin = IMX_GPIO_NR(4, 22);
+            gpio_direction_output(sdcard_power_pin , 0);
+			usdhc_cfg[0].sdhc_clk = mxc_get_clock(MXC_ESDHC_CLK);
 			break;
 		default:
 			printf("Warning: you configured more USDHC controllers"
@@ -234,27 +242,18 @@ void reset_phy()
 	 * be deasserted before 25ms have passed since the power supply has
 	 * reached 80% of the operating voltage. At this point of the code
 	 * we can assume the second premise is already accomplished.
+     *
+     * r1701 - v1 Board - MX6_PAD_LCD_DATA18__GPIO3_IO23 --> ENET1_RST Pin
+     *
 	 */
-	if (board_version == 1) {
-		int phy_reset_gpio = IMX_GPIO_NR(3, 2);
-
-		/* Assert PHY reset (low) */
-		gpio_direction_output(phy_reset_gpio , 0);
-		udelay(100);
-		/* Deassert PHY reset (high) */
-		gpio_set_value(phy_reset_gpio, 1);
-	} else {
-		/* MCA_IO7 is connected to PHY reset */
-		int reset = (1 << 7);
-
-		/* Configure as output */
-		mca_update_bits(MCA_CC6UL_GPIO_DIR_0, reset, reset);
-		/* Assert PHY reset (low) */
-		mca_update_bits(MCA_CC6UL_GPIO_DATA_0, reset, 0);
-		udelay(100);
-		/* Deassert PHY reset (high) */
-		mca_update_bits(MCA_CC6UL_GPIO_DATA_0, reset, reset);
-	}
+    int phy_reset_gpio = IMX_GPIO_NR(3, 23);
+
+    /* Assert PHY reset (low) */
+    gpio_direction_output(phy_reset_gpio , 0);
+    udelay(100);
+
+    /* Deassert PHY reset (high) */
+    gpio_set_value(phy_reset_gpio, 1);
 }

 int board_eth_init(bd_t *bis)
@@ -263,6 +262,11 @@ int board_eth_init(bd_t *bis)

 	setup_iomux_fec();

+    /* Turn on power to ethernet module */
+    int eth_power_gpio = IMX_GPIO_NR(3, 22);
+    gpio_direction_output(eth_power_gpio , 0);
+    udelay(100);
+
 	ret = fecmxc_initialize_multi(bis, CONFIG_FEC_ENET_DEV,
 		CONFIG_FEC_MXC_PHYADDR, IMX_FEC_BASE);
 	if (ret)
@@ -279,12 +283,12 @@ static int setup_fec(int fec_id)
 		= (struct iomuxc_gpr_base_regs *) IOMUXC_GPR_BASE_ADDR;
 	int ret;

-	if (board_version != 1) {
-		/* Enable PHY power */
-		int phy_power_gpio = IMX_GPIO_NR(3, 2);
-
-		gpio_direction_output(phy_power_gpio , 1);
-	}
+    /*
+     * r1701 - v1 - ENET1_PWR* - MX6_PAD_LCD_DATA17__GPIO3_IO22, Active Low Signal.
+     */
+    /* Enable PHY power */
+    int phy_power_gpio = IMX_GPIO_NR(3, 22);
+    gpio_direction_output(phy_power_gpio , 0);

 	if (0 == fec_id) {
 		if (check_module_fused(MX6_MODULE_ENET1))
@@ -328,14 +332,15 @@ int board_phy_config(struct phy_device *phydev)
 #define UCTRL_PWR_POL		(1 << 9)

 static iomux_v3_cfg_t const usb_otg_pads[] = {
-	MX6_PAD_GPIO1_IO00__ANATOP_OTG1_ID | MUX_PAD_CTRL(OTG_ID_PAD_CTRL),
+	MX6_PAD_GPIO1_IO05__ANATOP_OTG2_ID | MUX_PAD_CTRL(OTG_ID_PAD_CTRL),
+	MX6_PAD_GPIO1_IO03__USB_OTG2_OC | MUX_PAD_CTRL(OTG_ID_PAD_CTRL),
+	MX6_PAD_GPIO1_IO02__GPIO1_IO02 | MUX_PAD_CTRL(OTG_ID_PAD_CTRL),
 };

 /* At default the 3v3 enables the MIC2026 for VBUS power */
 static void setup_usb(void)
 {
-	imx_iomux_v3_setup_multiple_pads(usb_otg_pads,
-					 ARRAY_SIZE(usb_otg_pads));
+	imx_iomux_v3_setup_multiple_pads(usb_otg_pads, ARRAY_SIZE(usb_otg_pads));
 }

 int board_usb_phy_mode(int port)
diff --git a/include/configs/r1701.h b/include/configs/r1701.h
index 8a8727408c..dd5261589c 100644
--- a/include/configs/r1701.h
+++ b/include/configs/r1701.h
@@ -221,9 +221,9 @@
 	""	/* end line */
 #endif

-#define CONFIG_SYS_MMC_ENV_DEV		0   /* USDHC2 */
+#define CONFIG_SYS_MMC_ENV_DEV		0   /* USDHC1 */
 #define CONFIG_SYS_MMC_ENV_PART		0	/* user area */
-#define CONFIG_MMCROOT			"/dev/mmcblk1p2"  /* USDHC2 */
+#define CONFIG_MMCROOT			"/dev/mmcblk1p2"  /* USDHC1 */

 /* Carrier board version in OTP bits */
 #define CONFIG_HAS_CARRIERBOARD_VERSION
--
2.11.0

