From 5c6a69e21a7073e731e2a3e79362ac926b2ac850 Mon Sep 17 00:00:00 2001
From: Devang Patel <dpatel@hach.com>
Date: Wed, 18 Apr 2018 13:46:03 -0600
Subject: [PATCH 3/3] Add Micrel Phy chip support

---
 include/configs/r1701.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/configs/r1701.h b/include/configs/r1701.h
index 53041690f5..8aa674b621 100644
--- a/include/configs/r1701.h
+++ b/include/configs/r1701.h
@@ -23,8 +23,9 @@
 /* #define CONFIG_CMD_BEE */
 
 /* Define PHY Chip. We have a different PHY chip from Micrel (KSZ8081RNB) compared to Digi's Microchip LAN8720 */
-#if defined(CONFIG_PHY_SMSC)
+#ifdef CONFIG_PHY_SMSC
 #undef CONFIG_PHY_SMSC
+#endif
 #define CONFIG_PHY_MICREL
 
 /* FLASH and environment organization */
-- 
2.11.0

